La'Tonia/super_function.py

```diff
@@ -0,0 +1,163 @@
+# basically Square class inherits area from Rectangle class
+class Square(Rectangle):
+    def __init__(self, length):
+        super().__init__(length, length)
+
+square = Square(4)
+square.area()
+print('PRINTS AREA OF SQUARE WITH VALUE 4 USING SUPER()')
+print(square.area())
+print()
+
+
+# why am I passing length twice - first length is for child class Square and second length is for parent/super/parent class length?
+class Square(Rectangle):
+    def __init__(self, length):
+        super().__init__(length, length)
+
+'''
+code below does the same as lines 65-67
+class Square(Rectangle):
+    def __init__(self, length):
+        super(Square, self).__init__(length, length)
+'''
+
+class Cube(Square):
+    def surface_area(self):
+        face_area = super().area()
+        return face_area * 6
+
+    def volume(self):
+        face_area = super().area()
+        return face_area * self.length
+
+'''
+code below does the same as lines 76-83
+class Cube(Square):
+    def surface_area(self):
+        face_area = super(Square, self).area()
+        return face_area * 6
+
+    def volume(self):
+        face_area = super(Square, self).area()
+        return face_area * self.length
+'''
+
+cube = Cube(3)
+cube.surface_area()
+print('PRINTS NUMBER THREE CUBED')
+print(cube.surface_area())
+print()
+
+cube.volume()
+print('PRINTS CUBE VOLUME')
+print(cube.volume())
+print()
+
+
+# illustrating multiple inheritance with triangle
+class Triangle:
+    def __init__(self, base, height):
+        self.base = base
+        self.height = height
+        super().__init__()
+
+    def tri_area(self):
+        return 0.5 * self.base * self.height
+
+    def area(self):
+        return 0.5 * self.base * self.height
+
+    # class RightPyramid(Triangle, Square):
+    #     def __init__(self, base, slant_height):
+    #         self.base = base
+    #         self.slant_height = slant_height
+
+    #     def area(self):
+    #         base_area = super().area()
+    #         perimeter = super().perimeter()
+    #         return 0.5 * perimeter * self.slant_height + base_area
+
+# inspect order per resolution needs - what happens and in what order
+    # RightPyramid.__mro__
+
+# code lines 118 to 126 becomes ...
+# but this whole bottom code block not working ... too ambitious at this point
+class RightPyramid(Square, Triangle):
+    def __init__(self, base, slant_height, **kwargs):
+        self.base = base
+        self.slant_height = slant_height
+        kwargs["height"] = slant_height
+        kwargs["length"] = base
+        super().__init__(self.base)
+        super().__init__(base, **kwargs)
+
+    def area(self):
+        base_area = super().area()
+        perimeter = super().perimeter()
+        return 0.5 * perimeter * self.slant_height + base_area
+
+    def area_2(self):
+        base_area = super().area()
+        triangle_area = super().tri_area()
+        return triangle_area * 4 + base_area
+
```

Comment:
These polygon classes would be interesting using some sort of visual output. Maybe return to this when we have free time, for class to print  
   0
 000
0000